### PR TITLE
Gpu and vector examples

### PR DESCRIPTION
**Pull request by linuxlonelyeagle**
Add some GPU and vector examples.

### CODE REVIEW

File: examples/MLIRGPU/vector-to-gpu.mlir [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  } 
+  gpu.module @kernels { 
+    gpu.func @kernel1(%arg0 : memref<16x16xf16>, %arg1 : memref<16x16xf16>, %arg2 : memref<16x16xf32>) kernel {
+        %f0 = arith.constant 0.0 : f16
+        %f0f32 = arith.constant 0.0 : f32
+        %c0 = arith.constant 0 : index
+        %arg_v0 = vector.transfer_read %arg0[%c0, %c0], %f0 : memref<16x16xf16>, vector<16x16xf16>
+        %arg_v1 = vector.transfer_read %arg1[%c0, %c0], %f0 : memref<16x16xf16>, vector<16x16xf16>
+        %arg_v2 = vector.transfer_read %arg2[%c0, %c0], %f0f32 : memref<16x16xf32>, vector<16x16xf32>
+        %result = vector.contract {indexing_maps = [#map0, #map1, #map2], iterator_types = ["parallel", "parallel", "reduction"], kind = #vector.kind<add>} %arg_v0, %arg_v1, %arg_v2 : vector<16x16xf16>, vector<16x16xf16> into vector<16x16xf32>
```
Comment (hanliutong) [inline examples/MLIRGPU/vector-to-gpu.mlir:46]: What's the meaning of `kind = #vector.kind<add>` here?

**Review by hanliutong — CHANGES_REQUESTED** [review]
Please add the empty line at the end of each files.

File: examples/MLIRGPU/async-execute.mlir [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    }
+    gpu.dealloc %mem0 : memref<100xi32>
+    gpu.dealloc %mem1 : memref<100xi32>
+    async.yield
+  }
+  async.await %t2 : !async.token
+  call @printMemrefI32(%mem_cast) : (memref<*xi32>) -> ()
+  return 
+}
+func.func private @printMemrefI32(%ptr : memref<*xi32>)
```
[the remark this answers is not in the document]
Comment (linuxlonelyeagle) [reply examples/MLIRGPU/async-execute.mlir]: I have added an empty line at the end of the file.

File: examples/MLIRGPU/vector-to-gpu.mlir (hunk 1 above)
Comment (linuxlonelyeagle) [reply examples/MLIRGPU/vector-to-gpu.mlir:46]: I may meet some problem.The mlir documentation show that An optional kind attribute may be used to specify the combining function between the intermediate result and accumulator argument of rank K. This attribute can take the values add/mul/min/max for int/fp, and/or/xor for int only. The default is “add”.But i change the add and use the mul, it doesn't work.I may need to spend some time on it.

**Review by hanliutong — APPROVED** [review]

**Comment by hanliutong** [comment]
LGTM

**Comment by zhanghb97** [comment]
Thanks! @linuxlonelyeagle @hanliutong

**Pull request merged**